### ISSUE TITLE
HiddenFieldCheck doesn't report violation when a class member is defined in static or non-static block

### ISSUE THREAD

**Issue by Vladlis** [checkstyle/checkstyle#1552]
Here is an example for a static block:
```bash
vlad@vlad-N61PC-M2S ~/cli $ java -jar checkstyle-6.8.1-all.jar -c my_check.xml Test.java
Starting audit...
Audit done.

vlad@vlad-N61PC-M2S ~/cli $ cat my_check.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
         <module name="HiddenFieldCheck"/>
    </module>
</module>

vlad@vlad-N61PC-M2S ~/cli $ cat Test.java
class Test {
    static {
        String bar;
    }
    void method(int bar) {
        bar++;
    }
}
```

**Comment by Vladlis** [comment]
My bad, <code>String bar</code> is not a class member and cannot be accessed from <code>void method(int bar)</code>
